packages/kit/src/module/install.ts

```diff
@@ -38,7 +38,7 @@ export async function installModule (moduleToInstall: string | NuxtModule, inlin
     nuxt.options.build.transpile.push(normalizeModuleTranspilePath(moduleToInstall))
     const directory = getDirectory(moduleToInstall)
     if (directory !== moduleToInstall && !localLayerModuleDirs.has(directory)) {
-      nuxt.options.modulesDir.push(directory)
+      nuxt.options.modulesDir.push(resolve(directory, '..'))
```

Comment:
'modulesDir' in this context is a place where we resolve node_modules from. The parent directory may not work, particularly in a hoisted context. I think this should likely stay unchanged here but perhaps we add a bit of additional logic in `template.ts`?